### ISSUE TITLE
com.hazelcast.jet.python.PythonServiceTest [HZ-1776] [HZ-4730]

### ISSUE THREAD

**Issue by sumnerib** [hazelcast/hazelcast#22885]
_master_ (commit 1cb1e05d2b84fdb0289f8d64b4c42bacb70b101e)

Failed on IbmJDK8-fips: https://jenkins.hazelcast.com/job/Hazelcast-master-IbmJDK8-fips-nightly/482/testReport/junit/com.hazelcast.jet.python/PythonServiceTest

Also fails for: https://jenkins.hazelcast.com/job/Hazelcast-master-IbmJDK8-fips-nightly/482/testReport/junit/com.hazelcast.jet.python/PythonInitCleanupTest/

<details><summary>Stacktrace:</summary>

```
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 1
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:386)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1958)
	at com.hazelcast.jet.Job.join(Job.java:78)
	at com.hazelcast.jet.python.PythonServiceTest.batchStage_mapUsingPython_onAllMembers(PythonServiceTest.java:126)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:825)
Caused by: com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 1
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:114)
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:109)
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:89)
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:49)
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:59)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$null$9c5eb802$1(ExecutionPlan.java:456)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:532)
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.lambda$initProcSuppliers$10beb6ff$1(ExecutionPlan.java:455)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1651)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.lang.Thread.run(Thread.java:825)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 1
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:105)
	... 16 more

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-IbmJDK8-fips-nightly/482/

**Comment by kwart** [comment]
Hopefully will be fixed by Python upgrade (3.6->3.8) in https://github.com/hazelcast/DevOps/commit/ee62c5068d5071497a19f3b46047a86218b3afc2

Verification run: https://jenkins.hazelcast.com/job/Hazelcast-master-IbmJDK8-fips-nightly/484/

**Comment by kwart** [comment]
The `PythonServiceTest` in the verification run passed. Closing.

**Comment by Patras3** [comment]
Failed again on _master_ with OpenJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK17-arm64/17/testReport/com.hazelcast.jet.python/PythonServiceTest/

**Comment by kwart** [comment]
Reassigned to Patryk, who manages the ARM64 environment for the Corretto17 tests.
The issue is caused by the old Python version (`3.7.16`) in the AMI (`arm64-mvn-corretto17-cmake`). Our Python Protobuf dependency requires at least Python `3.8`.

(Currently, we have the Python version 3.10.12 in the JDK compatibility Docker image).

Test output contains:
```
02:11:01,550 DEBUG || - [python] python-init-logger_29449 - [127.0.0.1]:5701 [dev] [5.5.0-SNAPSHOT] + flock --exclusive /home/ec2-user/.jet-pip.lock python3 -m pip install protobuf==4.25.3 grpcio==1.60.0
02:11:02,198 DEBUG || - [python] python-init-logger_29449 - [127.0.0.1]:5701 [dev] [5.5.0-SNAPSHOT] ERROR: Ignored the following yanked versions: 3.18.0, 4.0.0rc1, 4.0.0rc2, 4.21.0
02:11:02,198 DEBUG || - [python] python-init-logger_29449 - [127.0.0.1]:5701 [dev] [5.5.0-SNAPSHOT] ERROR: Ignored the following versions that require a different python version: 4.25.0 Requires-Python >=3.8; 4.25.0rc1 Requires-Python >=3.8; 4.25.0rc2 Requires-Python >=3.8; 4.25.1 Requires-Python >=3.8; 4.25.2 Requires-Python >=3.8; 4.25.3 Requires-Python >=3.8; 5.26.0 Requires-Python >=3.8; 5.26.0rc1 Requires-Python >=3.8; 5.26.0rc2 Requires-Python >=3.8; 5.26.0rc3 Requires-Python >=3.8; 5.26.1 Requires-Python >=3.8
```

**Comment by Patras3** [comment]
I've fixed arm64 builds by updating Python to 3.8 version.
The remaining work here is only for soaks-tests.

**Comment by Patras3** [comment]
I've fixed arm64 builds by updating Python to 3.8 version.
The remaining work here is only for soaks-tests and will be tracked in Jira: https://hazelcast.atlassian.net/browse/HZ-4730
